falcon/asgi/app.py

```diff
@@ -474,13 +486,21 @@ async def __call__(self, scope, receive, send):  # noqa: C901
                 if not event:
                     event = SSEvent()
 
+                # NOTE(kgriffs): According to the ASGI spec, once the client
+                #   disconnects, send() acts as a no-op. We have to check
+                #   the connection state using watch_disconnect() above.
                 await send({
-                    'type': 'http.response.body',
+                    'type': EventType.HTTP_RESP_BODY,
                     'body': event.serialize(),
                     'more_body': True
                 })
 
-            await send({'type': 'http.response.body'})
+                if watcher.done():
+                    break
+
+            watcher.cancel()
```

Comment:
I forgot to await the watcher after cancelling it... will fix this in the next round.